src/ComponentManager.ts

```diff
@@ -45,78 +98,113 @@ export default class ModuleLoader {
     this.bot.on("message", this.onMessage.bind(this));
   }
 
+  private ignores: string[] = ["params.ts", "KeyValue.ts"]
   private onBot() {
     const fs = require('fs');
-
-    this.modules["ComponentManager.ts"] = this.cmCommands;
+    this.modules2.Add("ComponentManager", this.managerCommands);
+    this.iModules.Add("ComponentManager", new KeyValueArray<Command>());
+    // console.log("- ComponentManager")
+    for(let mod in this.managerCommands) {
+      // console.log(mod)
+      this.iModules.Item("ComponentManager").Add(mod, this.oldparamss[mod]);
```

Comment:
`oldparamms` doesn't seem to exist anywhere else in the file, and the compiler is whining about this too.